src/http/common.cc

```diff
@@ -124,6 +124,48 @@ output_stream<char> make_http_chunked_output_stream(output_stream<char>& out) {
     return output_stream<char>(http_chunked_data_sink(out), 32000, opts);
 }
 
+class http_content_length_data_sink_impl : public data_sink_impl {
+    output_stream<char>& _out;
+    const size_t _limit;
+    size_t& _bytes_written;
+
+public:
+    http_content_length_data_sink_impl(output_stream<char>& out, size_t& len)
+            : _out(out)
+            , _limit(std::exchange(len, 0))
+            , _bytes_written(len)
+    {
+    }
+    virtual future<> put(net::packet data)  override { abort(); }
+    using data_sink_impl::put;
+    virtual future<> put(temporary_buffer<char> buf) override {
+        if (buf.size() == 0 || _bytes_written == _limit) {
+            return make_ready_future<>();
+        }
+        auto size = std::min(buf.size(), _limit - _bytes_written);
+        return _out.write(buf.get(), size).then([this, size] {
+            _bytes_written += size;
+        });
+    }
+    virtual future<> close() override {
```

Comment:
shouldn't this return `_out.close()`?

Reply:
No, `_out` belongs to the underlying socket which may still be in use. This sink is just to put not more than the given amount of body bytes into the socket

Reply:
so please add a comment documenting that for future generations